electronics_lib/Passives.py

```diff
@@ -314,6 +297,222 @@ def derated_capacitance(row: Dict[str, Any]) -> Tuple[float, float]:
+        num_caps * capacitance[0],
+        num_caps * capacitance[1],
+      ))
+      self.assign(self.selected_capacitance, (
+        num_caps * min_nominal_capacitance[0],
+        num_caps * min_nominal_capacitance[1],
+      ))
+
+      cap_model = DummyCapacitor(capacitance=(self.single_cap_max, self.single_cap_max),
+                                      voltage=self.voltage)
+      self.c = ElementDict[DummyCapacitor]()
+      for i in range(num_caps):
+        self.c[i] = self.Block(cap_model)
+        self.connect(self.c[i].pos, self.pos)
+        self.connect(self.c[i].neg, self.neg)
+    else:
+      self.assign(self.selected_derated_capacitance, capacitance)
+      self.assign(self.selected_capacitance, min_nominal_capacitance)
+
+      self.footprint(
+        'C', valid_footprint_spec,
+        {
+          '1': self.pos,
+          '2': self.neg,
+        },
+        value=f'{UnitUtils.num_to_prefix(value, 3)}F'
+      )
+
+class DummyCapacitor(DummyDevice, Capacitor, FootprintBlock, GeneratorBlock):
+
+  """
+  Capacitor that does not derate, used for splitting a generic capacitor into multiple when desired capacitance is too high
+  """
+
+  @init_in_parent
+  def __init__(self, *args, **kwargs):
+    super().__init__(*args, **kwargs)
+
+    self.footprint_spec = self.Parameter(StringExpr(""))
+
+    # Default to be overridden on a per-device basis
+    self.single_nominal_capacitance = self.Parameter(RangeExpr((0, (22e-6)*1.25)))  # maximum capacitance in a single part
+
+    self.generator(self.select_capacitor, self.capacitance, self.voltage, self.single_nominal_capacitance,
+                   self.part_spec, self.footprint_spec)
+
+    # Output values
+    self.selected_capacitance = self.Parameter(RangeExpr())
+    self.selected_derated_capacitance = self.Parameter(RangeExpr())
+    self.selected_voltage_rating = self.Parameter(RangeExpr())
+
+  PACKAGE_MIN_CAP = {
+    'Capacitor_SMD:C_0603_1608Metric': 1e-7,
+    'Capacitor_SMD:C_0805_2012Metric': 1.1e-6,
+    'Capacitor_SMD:C_1206_3216Metric': 11e-6
+  }
+
+  def select_capacitor(self, capacitance: RangeVal, voltage: RangeVal,
+                       single_nominal_capacitance: RangeVal,
+                       part_spec: str, footprint_spec: str) -> None:
+
+    # avg_cap = (capacitance[0] + capacitance[1]) / 2
+    # for package_name in sorted(self.PACKAGE_MIN_CAP.keys())[::-1]:
+    #   if avg_cap > self.PACKAGE_MIN_CAP[package_name]:
+    #     self.calculated_package_size = package_name
+    #     break
+
+    # if not (footprint_spec == ""):
+    #   self.calculated_package_size = footprint_spec
+
+    self.assign(self.selected_capacitance, capacitance)
+    self.assign(self.selected_derated_capacitance, capacitance)
+
+    self.footprint(
+      'C', sorted(self.PACKAGE_MIN_CAP.keys())[-1], # TODO fix to a parameter footprint
+      {
+        '1': self.pos,
+        '2': self.neg,
+      },
+      value=f'{UnitUtils.num_to_prefix(capacitance[0], 3)}F'
+    )
```

Comment:
I think this could be vastly simplified by making it the responsibility of the calling function to select a package and capacitance, and all this does is create a footprint with the specified package and value without any additional logic.
(aside from perhaps creating the value string with proper sig figs in the footprint itself)